src/gui/editorwidgets/core/qgseditorwidgetautoconf.cpp

```diff
@@ -0,0 +1,110 @@
+/***************************************************************************
+    qgseditorwidgetautoconf.cpp
+    ---------------------
+    begin                : July 2016
+    copyright            : (C) 2016 by Patrick Valsecchi
+    email                : patrick.valsecchi at camptocamp.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgseditorwidgetautoconf.h"
+#include "qgseditorwidgetregistry.h"
+
+/** \ingroup gui
+ * Widget auto conf plugin that guesses what widget type to use in function of what the widgets support.
+ *
+ * @note not available in Python bindings
+ * @note added in QGIS 3.0
+ */
+class FromFactoriesPlugin: public QgsEditorWidgetAutoConfPlugin
```

Comment:
If these are non public api, it's better to wrap them in

    ///@cond PRIVATE

And 

    ///@endcond

That avoids them being included in the api docs